collector/src/main.rs

```diff
@@ -1141,9 +1166,102 @@ fn main_result() -> anyhow::Result<i32> {
 
             Ok(0)
         }
+        Commands::Download(cmd) => {
+            let target_dir = get_downloaded_crate_target(&benchmark_dir, &cmd);
+            check_target_dir(&target_dir, cmd.overwrite)?;
+
+            match cmd.command {
+                DownloadSubcommand::Git { url } => download_from_git(&target_dir, &url)?,
+                DownloadSubcommand::Crate { krate, version } => {
+                    download_from_crates_io(&target_dir, &krate, &version)?
+                }
+            };
+            println!("Benchmark stored at {}", target_dir.display());
+            Ok(0)
+        }
     }
 }
 
+fn check_target_dir(target_dir: &Path, overwrite: bool) -> anyhow::Result<()> {
+    if target_dir.exists() {
+        if overwrite {
+            std::fs::remove_dir_all(&target_dir).expect(&format!(
+                "Cannot remove previous directory at {}",
+                target_dir.display()
+            ));
+        } else {
+            return Err(anyhow::anyhow!(
+                "Directory {} already exists",
+                target_dir.display()
+            ));
+        }
+    }
+    Ok(())
+}
+
+fn get_downloaded_crate_target(benchmark_dir: &Path, cmd: &DownloadCommand) -> PathBuf {
+    let name = cmd.name.clone().unwrap_or_else(|| match &cmd.command {
+        DownloadSubcommand::Git { url } => url
+            .trim_end_matches("/")
```

Comment:
Please add a comment here explaining what the URL typically looks like, because that will make this string manipulation code much easier to understand.